src/collectors/github_runners.rs

```diff
@@ -0,0 +1,180 @@
+use crate::{http, http::is_token_flagged, Config};
+use anyhow::Error;
+use log::{debug, error, warn};
+use prometheus::core::Desc;
+use prometheus::proto::MetricFamily;
+use prometheus::{core::Collector, IntGauge, Opts};
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+use tokio::time::Duration;
+
+const GH_RUNNERS_ENDPOINT: &str = "https://api.github.com/repos/{owner_repo}/actions/runners";
+
+#[derive(Debug, serde::Deserialize)]
+struct ApiResponse {
+    total_count: usize,
+    runners: Vec<Runner>,
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct Runner {
+    id: usize,
+    name: String,
+    os: String,
+    status: String,
+    busy: bool,
+    labels: Vec<Label>,
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct Label {
+    id: usize,
+    name: String,
+    #[serde(rename = "type")]
+    the_type: String,
+}
+
+#[derive(Clone)]
+pub struct GithubRunners {
+    //api token to use
+    token: String,
+    // repos to track gha runners
+    repos: Vec<String>,
+    // metric namespace
+    ns: String,
+    // actual metrics
+    metrics: Arc<RwLock<Vec<IntGauge>>>,
+    // default metric description
+    desc: Desc,
+}
+
+impl GithubRunners {
+    pub async fn new(config: &Config) -> Result<Self, Error> {
+        let token = config.rust_runners_token.to_string();
+        let repos: Vec<String> = config
+            .gha_runners_repos
+            .split(',')
+            .map(|v| v.trim().to_string())
+            .collect();
+
+        let ns = String::from("gha_runner");
```

Comment:
Is there a reason why this is stored in the struct instead of being a constant?

Reply:
I've refactor that. we should have a `Desc` type field in each collector with all the metric's information and should read from there. (?)

I found out that the prometheus crate we use will hash all the metrics information (namespace and labels) and will error out if the same hash is already registered. 

We didn't had that issue before cos we only had one collector (that I believe should be changed too - we shoudn't be returning an empty vec from the `desc` method of the trait here https://github.com/rust-lang/monitorbot/blob/master/src/collectors/github_rate_limit.rs#L151-L154)

Can you have a look the way i'm doing in in this PR (for the runners collector) and tell me if you consider it resonable ?

Reply:
This looks reasonable, yeah!

Reply:
Great, so I'd like to open a new github `issue` to refactor the `rate_limit` collector regarding this issue about the `Desc` field in a follow up PR as soon as we get this one done.

Reply:
issue here : https://github.com/rust-lang/monitorbot/issues/18